http/vertx-web-validation/src/main/java/io/quarkus/ts/vertx/web/validation/ValidationHandlerOnRoutes.java

```diff
@@ -0,0 +1,150 @@
+import io.vertx.ext.web.validation.ValidationHandler;
+import io.vertx.ext.web.validation.builder.Bodies;
+import io.vertx.ext.web.validation.builder.Parameters;
+import io.vertx.ext.web.validation.builder.ValidationHandlerBuilder;
+import io.vertx.json.schema.SchemaParser;
+import io.vertx.json.schema.SchemaRouter;
+import io.vertx.json.schema.SchemaRouterOptions;
+import io.vertx.json.schema.common.dsl.ObjectSchemaBuilder;
+
+public class ValidationHandlerOnRoutes {
+    //TODO when Quarkus use vert.x version 4.4.6 we can use SchemaRepository instead of SchemaParser with SchemaRouter
+    //private SchemaRepository schemaRepository =SchemaRepository.create(new JsonSchemaOptions().setDraft(Draft.DRAFT7).setBaseUri(BASEURI));
+    private SchemaParser schemaParser;
+    private SchemaRouter schemaRouter;
+
+    @Inject
+    Vertx vertx;
+
+    private static ShopResource shopResource = new ShopResource();
+
+    private static final String ERROR_MESSAGE = "{\"error\": \"%s\"}";
+    private static final String SHOPPINGLIST_NOT_FOUND = "Shopping list not found in the list or does not exist with that name or price";
+
+    @PostConstruct
+    void initialize() {
+        schemaParser = createSchema();
+    }
+
+    private SchemaParser createSchema() {
+        schemaRouter = SchemaRouter.create(vertx, new SchemaRouterOptions());
+        schemaParser = SchemaParser.createDraft7SchemaParser(schemaRouter);
+        return schemaParser;
+    }
+
+    public void validateHandlerSoppingList(@Observes Router router) {
+        AtomicReference<String> queryAnswer = new AtomicReference<>();
+        router.get("/filterList")
+                .handler(ValidationHandlerBuilder
+                        .create(schemaParser)
+                        .queryParameter(param("shoppingListName", stringSchema()))
+                        .queryParameter(param("shoppingListPrice", numberSchema().with(maximum(100)))).build())
+                .handler(routingContext -> {
+                    RequestParameters parameters = routingContext.get(ValidationHandler.REQUEST_CONTEXT_KEY);
+                    String shoppingListName = parameters.queryParameter("shoppingListName").getString();
+                    Double totalPrice = parameters.queryParameter("shoppingListPrice").getDouble();
+
+                    // Logic to list shoppingList based on shoppingListName and totalPrice
+                    String shoppingListFound = fetchProductDetailsFromQuery(shoppingListName, totalPrice);
+                    queryAnswer.set(shoppingListFound);
+
+                    if (queryAnswer.get().equalsIgnoreCase(SHOPPINGLIST_NOT_FOUND)) {
+                        routingContext.response().setStatusCode(HttpResponseStatus.NOT_FOUND.code());
+                    }
+
+                    routingContext.response().putHeader(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN).end(queryAnswer.get());
+                }).failureHandler(routingContext -> {
```

Comment:
this failure handler results in hanging for other exceptions, you need to call `routingContext.next()`

Reply:
in the else statement do you mean?
`else {
                      routingContext.next();  
                    }`

Reply:
yes